wg-serving/proposals/llm_instance_gateway/README.md

```diff
@@ -0,0 +1,217 @@
+
+## Summary
+
+As presented in the [demo](https://youtu.be/NUBZg_uqqXk?si=v681EeYdGUGEVqQQ&t=1458) and building further upon the [joint proposal](https://docs.google.com/document/d/1BkwDlgFxSKKPHhM9kS28CdDIyJ3Xkdue3Iw1INaUkGw/edit?tab=t.0#heading=h.ajlsibmfh8wr), we are proposing that a gateway, focused on
+multiplexing 
+use cases upon shared hardware has distinct advantages in enabling efficient and fair use of multiple use-cases over a shared pool of compute.
+
+## Motivation
+
+Novel advancements in fine-tuning like [LoRA](https://arxiv.org/abs/2106.09685) and [Multi-LoRA](https://arxiv.org/abs/2310.18547) have enabled
+multiple distinct use cases to share accelerators. As this new tech is adopted,
+the Day1/2 operational 
+concerns quickly become necessary.
+Kubernetes as long been a standard in easing and automating operational tasks of
+workloads. A mechanism (gateway) within the K8s ecosystem is a 
+reasonable, and expected way for a user to support multiple LLM use cases on shared
+accelerators.
+
+### Goals
+
+#### Proposal Goals
+
+-   Create an Inference Gateway project group for wg-serving collaboration,
+    including: chat channel & dedicated repo (sponsored by sig-network)
+
+#### Gateway Goals
+
+-   Fast reconfiguration - New use cases (including LoRA adapters 
+or client    configuration) can be rolled out / back in seconds to clients without
+    
+    waiting for a new 
+  model server to start.
+-  Efficient accelerator sharing - Use cases can use less than an accelerator
+    or temporarily burst without needing to start a new model server leading to
+    fewer wasted accelerators and better pooling of shared capacity.
+-   Operational resilience - Use cases share available accelerators fairly and
+    can have distinct priorities, latency objectives, and failure policies.
+-   Standardized LoRA - Simple recommended patterns for deploying and loading
+    LoRA adapters on a wide range of Kubernetes environments into model servers.
+-   Composability - Approach should be composable with:
+    -   K8s Gateway 
+    API
+  -   Other gateway features and projects, including high level LLM gateways
+    -   existing deployment tools like kserve or kaito
+    
+    -   different model servers
+
+### Non-Goals
+
+
+#### Proposal Non-Goals
+-   Creation of a fully realized KEP
+
+#### Gateway Non-Goals
+
+-   Replacing the features of pre-existing Gateways
+-   Defining how serving workloads must be deployed
+
+## Proposal
+
+### Gateway
+
+#### CRD(s)
+
+To adequately achieve the above goals, we propose the addition of 1 or more CRDs
+to express:
+
+-   The boundaries of a compute pool that shares a base model
+    -   Including the deployment of a routing solution (PoC details below)
+-   A specific use case upon one or more backend pools
+    -   The objectives that this use case needs to achieve
+
+The example API we showed in our demo looked like:
+
+```
+kind: LLMRoute
+apiVersion: inference.x-k8s.io/v1alpha1
```

Comment:
do we have a section to talk about relationship with envoy gateway API? should we consider the similar group or not?

Reply:
Do we have an associated issue open on https://github.com/kubernetes-sigs/gateway-api as a x-ref?

Reply:
cc: @robscott

Reply:
The project proposal should focus on approval of the concept.  API design is not part of this project proposal, merely to suggest a starting point for the discussion (as discussed today at wg-serving, API and architecture proposals would be the first two artifacts in the new repo).

If we need to add additional language to the proposal to make that clear, let's do so.